modules/nf-core/windowmasker/mk_counts/main.nf

```diff
@@ -0,0 +1,43 @@
+process WINDOWMASKER_MKCOUNTS {
+    tag "$meta.id"
+    label 'process_low'
+
+    conda "bioconda::blast=2.13.0"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/blast:2.13.0--hf3cf87c_0':
+        'quay.io/biocontainers/blast:2.13.0--hf3cf87c_0' }"
+
+    input:
+    tuple val(meta), path(ref)
+
+    output:
+    tuple val(meta), path("*.txt")  , emit: counts
+    path "versions.yml"             , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args    = task.ext.args     ?: ""
+    def prefix  = task.ext.prefix   ?: "${meta.id}"
+
+    def memory = 3072
+    if (!task.memory) {
+        log.info '[WINDOWMASKER: MK_COUNTS] Available memory not known - defaulting to 3GB. Specify process memory requirements to change this.'
+    } else {
+        memory = (task.memory.toMega()*0.8).intValue()
+    }
+
+    """
+    windowmasker -mk_counts \\
+        $args \\
+        -mem ${memory} \\
+        -in ${ref} \\
+        -out ${prefix}.txt
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        windowmasker: \$(windowmasker -version-full | head -n 1)
+    END_VERSIONS
+    """
```

Comment:
Can you also add a stub section to both modules?

Reply:
Done. Stub is just empty output right?

```
    stub:
    """
    mkdir windowmasker
    touch windowmasker/test.inteval
    """

    stub:
    """
    mkdir windowmasker
    touch windowmasker/test.txt
    """
```

Reply:
Yes indeed!!! We also try to mimic the output generated with all the options given to the module (so in your case change the output extension for example)

Reply:
Can you also add the versions.yml creation to the stub? This can be the full code as it is in the main module (since it's a fairly simple calculation)

Reply:
Done. Thanks for the help over on Slack.

Reply:
You are welcome! 🥳